### ISSUE TITLE
Add an oldAddress param to AddressSet event

### ISSUE THREAD

**Issue by laizy** [ethereum-optimism/optimism#625]
**Is your feature request related to a problem? Please describe.**

Currently, the AddressSet event is defined as:
``` solidity
 event AddressSet(
        string _name,
        address _newAddress
    );
```
But the data-transport-layer service wants the old address, so have to do some searching to find the previous address set event,
which makes the code more complex and is not efficient enough. 

**Describe the solution you'd like**
Add oldAddress param to the events.
```solidity
 event AddressSet(
        string _name,
        address _newAddress,
       address _oldAddress
    );
```

**Comment by smartcontracts** [comment]
100% agree with this. I had a PR that made this exact change in our old contracts repo: https://github.com/ethereum-optimism/contracts/pull/224/files. We should make this change ASAP.

**Comment by smartcontracts** [comment]
Closed by #820, will be released in 0.4.0
